tests/test_degrade.py

```diff
@@ -380,6 +380,35 @@ def test_degrade_map_int_sum(self):
 
         testing.assert_almost_equal(deg_map, new_map.generate_healpix_map())
 
+    def test_degrade_map_wavg(self):
+        """
+        Test HealSparse.degrade functionality with float quantities
+        """
+        random.seed(12345)
+        nside_coverage = 32
+        nside_map = 1024
+        nside_new = 256
+        full_map = random.random(hp.nside2npix(nside_map))
+        weights = 0.25*np.ones_like(full_map)
```

Comment:
I think we need to test with weights that vary.  If they are all the same, this is completely equivalent to an unweighted mean.  And that's what tested below, with the ud_grade (unweighted).  And I guess this tests that the code path is valid, it doesn't actually test that this is doing the weights correctly (e.g., we could change the wmean to mean and it'll give the same answer and still pass tests.)

Reply:
right! This was me being lazy 😛.